README.md

```diff
@@ -207,23 +220,273 @@ More information can be found [here](http://www.diveintopython3.net/strings.html
+* The Python package is available on Python search path
+
+A Python package may have multiple modules, each with its own namespace. Each module has its own functions, classes, and variables. Each function or class has its own local namespace. In R, all functions and classes within the same package share a package namespace. To avoid the namespace collisions and allow customization to the wrapping package, we provide the following functions to help you pick which modules, functions, and classes to expose in R:
+
+* `generateRWrappers`
+* `generateRdFiles`
+
+The .Rd file generation must happen at the time the package is built, not at the time the package is loaded. The R function and constructor wrappers must happen at the time the package is loaded. These functions should be called with the similar parameters to ensure that all R wrapped functions have corresponding reference documentation.
+
+## Examples:
+
+In this example, I will demonstrate how we generate [synapser](https://github.com/Sage-Bionetworks/synapser) and [synapserutils](https://github.com/Sage-Bionetworks/synapserutils) packages by wrapping Python package [synapsePythonClient](https://github.com/Sage-Bionetworks/synapsePythonClient).
+
+The `synapseutils` module in `synapsePythonClient` package has the following structure:
+```
+    module: synapseutils
+        module: copy
+            function: copy
+            function: copyWiki
+            function: copyFileHandles
+        module: sync
+            function: syncToSynapse
+            function: syncFromSynapse
+        module: monitor
+            function: notifyMe
+```
+
+### Expose all functions and classes within a Python module
+
+#### Generate .Rd files
+
+To generate the R package `synapserutils`, our first attempt is to exposing all functions under `synapseutils` Python module. In `.configure` file, we execute an R script that calls `generateRdFiles` as following:
+```r
+generateRdFiles(srcRootDir,
+                pyPkg = "synapseutils",
+                module = "synapseutils")
+```
+
+Where:
+* `srcRootDir` is the path to `synapserutils` directory. The directory must exist prior to this call. `generateRdFiles` will create a folder `auto-man` and write the generated .Rd files in this folder.
+* `pyPkg` is the name of the Python package that needs to be imported, and
+* `module` is the name of the Python module to be wrapped.
+
+#### Generate R wrappers
+
+To generate R wrappers for Python functions and constructors, we need to add an `.onLoad` hook in the R package. This hook is defined in `synapserutils/R/zzz.R` file. 
+
+In the `.onLoad` function, we add the following:
+```r
+# For the R wrappers to be available in `synapserutils` package namespace,
+# `setGeneric` must be defined in the `synapserutils` package. Therefore,
+# we need to define it in `synapserutils` and pass it through `generateRWrappers`.
+callback <- function(name, def) {
+  setGeneric(name, def)
+}
+generateRWrappers(pyPkg = "synapseutils",
```

Comment:
perhaps it would help to show generateRWrappers() as being called from .onLoad, e.g.:
.onLoad<-function(<what goes here???>) {
  ...
  generateRWrappers(...)
  ...
}